Mathlib/LinearAlgebra/Finsupp/Supported.lean

```diff
@@ -183,6 +183,10 @@ theorem disjoint_supported_supported_iff [Nontrivial M] {s t : Set α} :
     (f : s →₀ M) : (supportedEquivFinsupp (R := R) s).symm f = f.extendDomain := by
   convert restrictSupportEquiv_symm_apply_coe ..
 
+@[simp] theorem supportedEquivFinsupp_symm_single (s : Set α) (i : s) (a : M) :
+    ((supportedEquivFinsupp (R := R) s).symm (single i a) : α →₀ M) = single ↑i a := by
```

Comment:
Can you prove this for `restrictSupportEquiv` too?

Reply:
`simp` already solves this by `simp only [restrictSupportEquiv_symm_apply_coe, extendDomain_single]`.

Reply:
It looks like this inconsistency is solely caused by classical instances. Namely, `restrictSupportEquiv_symm_apply_coe` is auto generated by the `@[simps]` call, so it happily automatically assumes the classical decidable instance that was supplied into the definition; whereas `supportedEquivFinsupp_symm_apply_coe` takes a custom DecidablePred instance.

Reply:
In that case, `restrictSupportEquiv_symm_apply_coe` is a bad lemma I guess

Reply:
Fix the statement of `restrictSupportEquiv_symm_apply_coe` by writing it manually.

Reply:
Done.